ci/do_coverity_local.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+#
+#  do_coverity_local.sh
+#
+#  This script builds Envoy with the Coverity Scan Built Tool.
+#
+#  It expects the following environment variables to be set:
+#    COVERITY_TOKEN      - set to the user's Coverity Scan project token.
+#    COVERITY_USER_EMAIL - set to the email address used with the Coverity account.
+#                          defaults to the local git config user.email.
+
+
+set -e
+
+. ./ci/envoy_build_sha.sh
+
+[[ -z "${ENVOY_DOCKER_BUILD_DIR}" ]] && ENVOY_DOCKER_BUILD_DIR=/tmp/envoy-docker-build
+mkdir -p ${ENVOY_DOCKER_BUILD_DIR}
+
+TEST_TYPE="bazel.coverity"
+COVERITY_USER_EMAIL=${COVERITY_USER_EMAIL:-$(git config user.email)}
+COVERITY_OUTPUT_FILE="${ENVOY_DOCKER_BUILD_DIR}"/envoy/source/exe/envoy-coverity-output.tgz
+
+if [ -n "$COVERITY_TOKEN" ]
+then
+  pushd ${ENVOY_DOCKER_BUILD_DIR}
+  rm -rf cov-analysis
+  wget https://scan.coverity.com/download/linux64 --post-data "token=${COVERITY_TOKEN}&project=Envoy+Proxy" -O coverity_tool.tgz
+  tar xvf coverity_tool.tgz
+  mv cov-analysis-linux* cov-analysis
+  popd
+else
+  echo "ERROR: COVERITY_TOKEN is required to download and run Coverity Scan."
+  exit 1
+fi
+
+ci/run_envoy_docker.sh "ci/do_ci.sh ${TEST_TYPE}"
+
+if [ -z "$COVERITY_TOKEN" ]
+then
+  echo "COVERITY_TOKEN environment variable not set."
+  echo "Manually submit build result to https://scan.coverity.com/projects/envoy-proxy/builds/new."
+elif [[ $(find "${COVERITY_OUTPUT_FILE}" -type f -size +256M 2>/dev/null) ]]
```

Comment:
Out of curiosity what's the reasoning behind the 256M check?

Reply:
Arbitrary heuristic. Resulting artifact is 300-400MB. However, in the case of a failed build or if Coverity wasn't able to analyse the build properly it will still generate an the equivalent of an empty scan artifact on the order of 1MB. A check against this prevents the empty artifact from being submitted in error. Shouldn't be an issue in practice but was an issue while I figured out how to do this.

I'll add a comment explaining why this check exists. Alternatively, I'll delete it if you prefer.